### ISSUE TITLE
nullPointerException when linking react-native-webview

### ISSUE THREAD

**Issue by maxlew** [react-native-community/react-native-webview#198]
Hi there, I'm getting the following error when trying to run `react-native run-android` after installing `react-native-webview`. Not sure if I've missed a very obvious step or not. The same configuration works in iOS.

**Versions**
```
react-native-cli: 2.0.1
react-native: 0.57.7
react-native-webview: 2.14.0
```

**Error**
```
A problem occurred configuring project ':react-native-webview'.
java.lang.NullPointerException (no error message)
```

**Reproduction steps**
```
react-native init app
cd app
yarn add react-native-webview
react-native link react-native-webview
react-native run-android
```

**Debug log at crash**
```
14:11:18.960 [DEBUG] [com.android.build.gradle.internal.pipeline.TransformManager] ADDED TRANSFORM(debugAndroidTest):
14:11:18.960 [DEBUG] [com.android.build.gradle.internal.pipeline.TransformManager]      Name: mergeJniLibs
14:11:18.960 [DEBUG] [com.android.build.gradle.internal.pipeline.TransformManager]      Task: transformNativeLibsWithMergeJniLibsForDebugAndroidTest
14:11:18.960 [DEBUG] [com.android.build.gradle.internal.pipeline.TransformManager]      InputStream: OriginalStream{name=ext-libs-res-plus-native-restricted-copy, scopes=[EXTERNAL_LIBRARIES], contentTypes=[NATIVE_LIBS], fileCollection=configuration ':react-native-webview:debugAndroidTestRuntimeClasspath'}
14:11:18.960 [DEBUG] [com.android.build.gradle.internal.pipeline.TransformManager]      InputStream: OriginalStream{name=ext-libs-native, scopes=[EXTERNAL_LIBRARIES], contentTypes=[NATIVE_LIBS], fileCollection=configuration ':react-native-webview:debugAndroidTestRuntimeClasspath'}
14:11:18.961 [DEBUG] [com.android.build.gradle.internal.pipeline.TransformManager]      InputStream: OriginalStream{name=sub-projects-res-plus-native-restricted-copy, scopes=[SUB_PROJECTS], contentTypes=[NATIVE_LIBS], fileCollection=file collection}
14:11:18.961 [DEBUG] [com.android.build.gradle.internal.pipeline.TransformManager]      InputStream: OriginalStream{name=sub-projects-native, scopes=[SUB_PROJECTS], contentTypes=[NATIVE_LIBS], fileCollection=file collection}
14:11:18.961 [DEBUG] [com.android.build.gradle.internal.pipeline.TransformManager]      InputStream: OriginalStream{name=mergedJniFolder, scopes=[PROJECT], contentTypes=[NATIVE_LIBS], fileCollection=file collection}
14:11:18.961 [DEBUG] [com.android.build.gradle.internal.pipeline.TransformManager]      InputStream: OriginalStream{name=local-ndk-build, scopes=[PROJECT], contentTypes=[NATIVE_LIBS], fileCollection=file collection}
14:11:18.961 [DEBUG] [com.android.build.gradle.internal.pipeline.TransformManager]      OutputStream: IntermediateStream{scopes=[PROJECT, SUB_PROJECTS, EXTERNAL_LIBRARIES], contentTypes=[NATIVE_LIBS], fileCollection=file collection}
14:11:18.961 [DEBUG] [org.gradle.model.internal.registry.DefaultModelRegistry] Project :react-native-webview - Registering model element 'tasks.transformNativeLibsWithMergeJniLibsForDebugAndroidTest' (hidden = false)
14:11:18.961 [DEBUG] [org.gradle.model.internal.registry.DefaultModelRegistry] Project :react-native-webview - Registering model element 'tasks.javaPreCompileDebugAndroidTest' (hidden = false)
14:11:18.962 [DEBUG] [org.gradle.model.internal.registry.DefaultModelRegistry] Project :react-native-webview - Registering model element 'tasks.compileDebugAndroidTestJavaWithJavac' (hidden = false)
14:11:18.962 [INFO] [com.android.build.gradle.tasks.factory.JavaCompileConfigAction] Using incremental javac compilation for :react-native-webview debugAndroidTest.
14:11:18.971 [DEBUG] [org.gradle.internal.progress.DefaultBuildOperationExecutor] Completing Build operation 'Configure project :react-native-webview'
14:11:18.971 [null] [org.gradle.internal.progress.DefaultBuildOperationExecutor]
14:11:18.971 [DEBUG] [org.gradle.internal.progress.DefaultBuildOperationExecutor] Completing Build operation 'Configure build'
14:11:18.972 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]
14:11:18.972 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] FAILURE: Build failed with an exception.
14:11:18.972 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]
14:11:18.972 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] * What went wrong:
14:11:18.972 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] A problem occurred configuring project ':react-native-webview'.
14:11:18.972 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] > java.lang.NullPointerException (no error message)
14:11:18.972 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]
14:11:18.972 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] * Try:
14:11:18.972 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] Run with --stacktrace option to get the stack trace.  Run with --scan to get full insights.
14:11:18.972 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]
14:11:18.972 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] * Get more help at https://help.gradle.org
14:11:18.972 [ERROR] [org.gradle.internal.buildevents.BuildResultLogger]
```

**Comment by Luftzig** [comment]
Happens to me too on versions =>2.8.0 (2.8.0 and later) of the library.

**Comment by Titozzz** [comment]
If you can reproduce this in a small project, I'll take a look

**Comment by lapto** [comment]
Sounds like a kotlin version conflict. 

in your android/build.gradle file, do you have a dependency on kotlin? If so, try changing it to the same version as the one used in this project (1.2.71). 
I belive this was added in the 2.8.1 with the refactor of the events

`buildscript {
    ext.kotlin_version = '1.2.71'
    repositories {
        google()
        jcenter()
        maven {
            url 'https://maven.fabric.io/public'
        }
    }
    dependencies {
        classpath 'com.android.tools.build:gradle:3.2.1'
        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
    }
}`

**Comment by Overtorment** [comment]
Can confirm issue with latest  `react-native-webview` and clean project after init.
The only workaround for me is to downgrade `npm install --save react-native-webview@v2.8.0`

**Comment by jamonholmgren** [comment]
![image](https://user-images.githubusercontent.com/1479215/50542122-d08ec580-0b69-11e9-9c8a-8669c492f780.png)

It seems to work for me with react-native-webview 2.14.3.

**Comment by Titozzz** [comment]
What is your java / android studio version please ?

**Comment by jordansexton** [comment]
I encountered this issue in our app and reproduced just now. I launched an existing AVD using:
```
$ANDROID_SDK/tools/emulator -wipe-data -avd Pixel_2_API_27
```

Steps from OP:

```
react-native init app
cd app
yarn add react-native-webview
react-native link react-native-webview
react-native run-android
```

#### Log
```
Starting JS server...
Building and installing the app on the device (cd android && ./gradlew installDebug)...
Starting a Gradle Daemon (subsequent builds will be faster)

> Configure project :app
WARNING: Configuration 'compile' is obsolete and has been replaced with 'implementation' and 'api'.
It will be removed at the end of 2018. For more information see: http://d.android.com/r/tools/update-dependency-configurations.html

> Configure project :react-native-webview
react-native-webview: using React Native sources from /Users/jordan/Work/Projects/Code/app/node_modules/react-native/android


FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':react-native-webview'.
> java.lang.NullPointerException (no error message)
```

#### Android Studio

```
Android Studio 3.2.1
Build #AI-181.5540.7.32.5056338, built on October 8, 2018
JRE: 1.8.0_152-release-1136-b06 x86_64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
macOS 10.14.1
```

#### Emulator
```
Android version: 8.1 (Oreo) - API 27
Emulator version: 28.0.22-5194135
```

I can also confirm that the issue doesn't occur when following the above commands with

```
yarn add react-native-webview@2.8.0
react-native run-android
```

#### Log
```
JS server already running.
Building and installing the app on the device (cd android && ./gradlew installDebug)...

> Configure project :app
WARNING: Configuration 'compile' is obsolete and has been replaced with 'implementation' and 'api'.
It will be removed at the end of 2018. For more information see: http://d.android.com/r/tools/update-dependency-configurations.html

> Task :react-native-webview:compileDebugJavaWithJavac
Note: /Users/jordan/Work/Projects/Code/app/node_modules/react-native-webview/android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: /Users/jordan/Work/Projects/Code/app/node_modules/react-native-webview/android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

> Task :app:transformNativeLibsWithStripDebugSymbolForDebug
Unable to strip library '/Users/jordan/Work/Projects/Code/app/android/app/build/intermediates/transforms/mergeJniLibs/debug/0/lib/armeabi/libicu_common.so' due to missing strip tool for ABI 'ARMEABI'. Packaging it as is.

> Task :app:installDebug
Installing APK 'app-debug.apk' on 'Pixel_2_API_27(AVD) - 8.1.0' for app:debug
Installed on 1 device.


BUILD SUCCESSFUL in 26s
48 actionable tasks: 48 executed
Running /Users/jordan/Library/Android/sdk/platform-tools/adb -s emulator-5554 reverse tcp:8081 tcp:8081
Starting the app on emulator-5554 (/Users/jordan/Library/Android/sdk/platform-tools/adb -s emulator-5554 shell am start -n com.app/com.app.MainActivity)...
Starting: Intent { cmp=com.app/.MainActivity }
```

**Comment by Titozzz** [comment]
Alright let me try that right now.

**Comment by Titozzz** [comment]
Can you please run 'java -version' and give me the output ? If it's 10 this is your issue and you need to use 8

**Comment by jordansexton** [comment]
```
➜ java -version
java version "10.0.2" 2018-07-17
Java(TM) SE Runtime Environment 18.3 (build 10.0.2+13)
Java HotSpot(TM) 64-Bit Server VM 18.3 (build 10.0.2+13, mixed mode)
```

Interesting. So I'm running the same Android Studio build and JRE as you then.

Why is Java 10 an issue? I'll try switching to 8.

**Comment by Titozzz** [comment]
Ok so I'm super glad, cause this is probably the issue. I don't know why java 10 fucks up, and since it was working before, we should investigate, maybe @Salakar would have an idea.

**Comment by Titozzz** [comment]
Just so you know: 
https://stackoverflow.com/questions/40108471/java-lang-nullpointerexception-no-error-message-apk-building
https://github.com/fastlane/fastlane/issues/12646
https://github.com/react-community/create-react-native-app/issues/603

**Comment by jordansexton** [comment]
Found this -- https://stackoverflow.com/questions/49484700/unable-to-run-react-native-project-due-to-java-version-10

Suggesting that updating to Gradle 4 might work. I'll try that as well, seems like it might be less annoying than downgrading Java.

**Comment by Titozzz** [comment]
on my personal project i use this gradle-wrapper.properties

```
#Thu Nov 15 12:11:26 CET 2018
distributionBase=GRADLE_USER_HOME
distributionPath=wrapper/dists
zipStoreBase=GRADLE_USER_HOME
zipStorePath=wrapper/dists
distributionUrl=https\://services.gradle.org/distributions/gradle-4.6-all.zip

```

**Comment by jordansexton** [comment]
Success! I used this _extremely_ lazy way to get to Gradle 4.6 on my project:

```
yarn add -D react-native-update-gradle
react-native link react-native-update-gradle
```

`react-native run-android` builds, and Android Studio no longer bugs me to update Gradle. Win-win.

**Comment by Titozzz** [comment]
TLDR: either use java 1.8 or update gradle

**Comment by xavieramoros** [comment]
I'm still having the "nullPointerException error" despite using: java 1.8 and gradle 4.7. 
Downgrading to react-native-webview 2.8.0 fixes the issue but it's like a temporary fix.

I'm using:
react-native 0.58.3
react-native-webview 5.2.1

android studio 3.1.1

java -version:
```
java version "1.8.0_161"
Java(TM) SE Runtime Environment (build 1.8.0_161-b12)
Java HotSpot(TM) 64-Bit Server VM (build 25.161-b12, mixed mode)
```

Things I tried without success:
-update android studio from 3.1.1 to 3.3.1
-specify jdk location in gradle.properties:
>org.gradle.java.home=/Library/Java/JavaVirtualMachines/jdk1.8.0_161.jdk/Contents/Home/

-downgrade gradlew from 4.7 to 4.6
-specifying kotlin_version in android/build.gradle:
```
buildscript {
    ext {
        buildToolsVersion = "28.0.3"
        minSdkVersion = 16
        compileSdkVersion = 28
        targetSdkVersion = 27
        kotlin_version = "1.2.71" 
    }
```
and adding to dependencies:
> classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"

-add to compileOptions:
```
compileOptions {
        sourceCompatibility JavaVersion.VERSION_1_8
        targetCompatibility JavaVersion.VERSION_1_8
}
```

Any more ideas?

**Comment by Titozzz** [comment]
I think with android studio 3.2 and more you need kotlin_version > 1.3

**Comment by lapto** [comment]
@[xavieramoros](/xavieramoros) Unless your using kotlin in your project I dont think you need to include:
```
kotlin_version = "1.2.71"
classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
``` 
In my case I have kotlin code that used an older version that caused a conflict when adding this package.

**Comment by xavieramoros** [comment]
Thanks for the kotlin clarification. 

I fixed my issue. I am using a "staging" buildType, I needed to add a matchingFallbacks in order to build correctly the app with assembleStaging.

Something like:
in android/app/build.gradle:
```
        debug {
            multiDexEnabled true
            applicationIdSuffix ".debug"
        }
        staging {
            multiDexEnabled true
            minifyEnabled enableProguardInReleaseBuilds
            proguardFiles getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro"
            signingConfig signingConfigs.release
            matchingFallbacks = ['debug'] <------ THIS FIXED MY ISSUE
            applicationIdSuffix ".staging"
        }
        release {
            minifyEnabled enableProguardInReleaseBuilds
            proguardFiles getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro"
            signingConfig signingConfigs.release
        }
```

**Comment by gbferreira** [comment]
This is a good tutorial to downgrade java:
https://stackoverflow.com/questions/46513639/how-to-downgrade-java-from-9-to-8-on-a-macos-eclipse-is-not-running-with-java-9

**Comment by andreiChis70** [comment]
What worked for me is running this before 'run-android':
`adb devices`

**Comment by Hmoulvad** [comment]
> Can confirm issue with latest `react-native-webview` and clean project after init.
> The only workaround for me is to downgrade `npm install --save react-native-webview@v2.8.0`

Yup it works to downgrade to 2.8
